src/targets/cpu/dnnl.cpp

```diff
@@ -67,7 +67,8 @@ dnnl::memory::data_type to_dnnl_memory_data_type(shape::type_t t)
     case st::float_type: return dt::f32;
     case st::int32_type: return dt::s32;
     case st::int8_type: return dt::s8;
-    case st::uint8_type: return dt::u8;
+    case st::uint8_type:
+    case st::fp8e4m3fnuz_type: return dt::u8;
```

Comment:
This seems wrong, `dt::u8` is an integer not a float. We should throw an error for fp8, when using zendnn. It should already be removed since we just whitelist the supported operators [here](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/targets/cpu/target.cpp#L69).

Reply:
fp8 is eliminated through eliminiate_dtype pass with converts to float.  
but later simplify_algebra is removing convert and turning test back to fp8. 

e.g. 

```
290 Validate ...
291
292 Pass: dead_code_elimination
293 module: "main"
294 z = @param:z -> fp8e4m3fnuz_type, {1, 2}, {2, 1}, target_id=0
295 y = @param:y -> fp8e4m3fnuz_type, {3, 2}, {2, 1}, target_id=0
296 x = @param:x -> fp8e4m3fnuz_type, {2, 2}, {2, 1}, target_id=0
297 @3 = convert[target_type=2](x) -> float_type, {2, 2}, {2, 1}, target_id=0
298 @4 = convert[target_type=2](y) -> float_type, {3, 2}, {2, 1}, target_id=0
299 @5 = convert[target_type=2](z) -> float_type, {1, 2}, {2, 1}, target_id=0
300 @6 = concat[axis=0](@3,@4,@5) -> float_type, {6, 2}, {2, 1}, target_id=0
301 @7 = convert[target_type=12](@6) -> fp8e4m3fnuz_type, {6, 2}, {2, 1}, target_id=0
302
303
304 Pass: simplify_algebra
305 z = @param:z -> fp8e4m3fnuz_type, {1, 2}, {2, 1}, target_id=0
306 y = @param:y -> fp8e4m3fnuz_type, {3, 2}, {2, 1}, target_id=0
307 x = @param:x -> fp8e4m3fnuz_type, {2, 2}, {2, 1}, target_id=0
308 @3 = concat[axis=0](x,y,z) -> fp8e4m3fnuz_type, {6, 2}, {2, 1}, target_id=0
309 @4 = convert[target_type=2](@3) -> float_type, {6, 2}, {2, 1}, target_id=0
310 @5 = convert[target_type=12](@4) -> fp8e4m3fnuz_type, {6, 2}, {2, 1}, target_id=0

```